pkg/util/gitutil/git.go

```diff
@@ -55,48 +90,119 @@ func GetGitRepository(dir string) (*git.Repository, error) {
 
 // GetGitHubProjectForOrigin returns the GitHub login, and GitHub repo name if the "origin" remote is
 // a GitHub URL.
-func GetGitHubProjectForOrigin(dir string) (string, string, error) {
+func GetGitHubProjectForOrigin(dir string) (*VCSInfo, error) {
 	repo, err := GetGitRepository(dir)
 	if repo == nil {
-		return "", "", fmt.Errorf("no git repository found from %v", dir)
+		return nil, fmt.Errorf("no git repository found from %v", dir)
 	}
 	if err != nil {
-		return "", "", err
+		return nil, err
 	}
-	return GetGitHubProjectForOriginByRepo(repo)
+	remoteURL, err := GetGitRemoteURL(repo, "origin")
+
+	if err != nil {
+		return nil, err
+	}
+	return TryGetVCSInfo(remoteURL)
 }
 
-// GetGitHubProjectForOriginByRepo returns the GitHub login, and GitHub repo name if the "origin" remote is
-// a GitHub URL.
-func GetGitHubProjectForOriginByRepo(repo *git.Repository) (string, string, error) {
-	remote, err := repo.Remote("origin")
+// GetGitRemoteURL returns the remote URL for the given remoteName in the repo.
+func GetGitRemoteURL(repo *git.Repository, remoteName string) (string, error) {
+	remote, err := repo.Remote(remoteName)
 	if err != nil {
-		return "", "", errors.Wrap(err, "could not read origin information")
+		return "", errors.Wrap(err, "could not read origin information")
 	}
 
 	remoteURL := ""
 	if len(remote.Config().URLs) > 0 {
 		remoteURL = remote.Config().URLs[0]
 	}
-	project := ""
 
-	const GitHubSSHPrefix = "git@github.com:"
-	const GitHubHTTPSPrefix = "https://github.com/"
-	const GitHubRepositorySuffix = ".git"
+	return remoteURL, nil
+}
 
-	if strings.HasPrefix(remoteURL, GitHubSSHPrefix) {
-		project = trimGitRemoteURL(remoteURL, GitHubSSHPrefix, GitHubRepositorySuffix)
-	} else if strings.HasPrefix(remoteURL, GitHubHTTPSPrefix) {
-		project = trimGitRemoteURL(remoteURL, GitHubHTTPSPrefix, GitHubRepositorySuffix)
+// IsGitOriginURLGitHub returns true if the provided remoteURL is detected as GitHub.
+func IsGitOriginURLGitHub(remoteURL string) bool {
+	return strings.Contains(remoteURL, GitHubHostName)
+}
+
+// TryGetVCSInfo attempts to detect whether the provided remoteURL
+// is an SSH or an HTTPS remote URL. It then extracts the repo, owner name,
+// and the type (kind) of VCS from it.
+func TryGetVCSInfo(remoteURL string) (*VCSInfo, error) {
```

Comment:
Given that it is just parsing a string, I don't think this function should return an `error` if it cannot identify the VCS host, and instead just return a `nil` value for `VCSInfo`.

Reply:
I kept the contract the same as the previous function, which I replaced with this one. The previous function returned an error and its callers explicitly ignored it.